snuba/query/expressions.py

```diff
@@ -261,12 +269,25 @@ def transform(self, func: Callable[[Expression], Expression]) -> Expression:
         one transforms the internal function before applying the function to the
         parameters.
         """
-        transformed = replace(
-            self,
-            internal_function=self.internal_function.transform(func),
-            parameters=tuple(map(lambda child: child.transform(func), self.parameters)),
+        replaced_internal = self.internal_function.transform(func)
+        replaced_params = tuple(
+            map(lambda child: child.transform(func), self.parameters)
         )
-        return func(transformed)
+        if (
+            replaced_internal != self.internal_function
+            or replaced_params != self.parameters
```

Comment:
Did you investigate how the number of replaced values impacts performance? Should we replace both if only one is different?

Reply:
Replacing one or both has minimal impact as long as we have to replace one. The goal here is to save the instantiation. In order to avoid the instantiation, both values have to be unchanged.